compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -329,7 +329,7 @@ object Interactive {
 
   def contextOfPath(path: List[Tree])(implicit ctx: Context): Context = path match {
     case Nil | _ :: Nil =>
-      ctx.run.runContext.fresh.setCompilationUnit(ctx.compilationUnit)
```

Comment:
@odersky Was this intentional? If so, what's the rational for using the run context?

Reply:
Pretty sure that's intentional. `contextOfPath` constructs a context with the correct scoping information. If you start from a random Context then you'll have too much stuff in the scope.

Reply:
I could patch the `rootContext` of the new run as we used to do:
```scala
class ReplCompiler extends Compiler {
  def newRun(initCtx: Context, state: State) = new Run(this, initCtx) {
    override protected[this] def rootContext(implicit ctx: Context) =
      addMagicImports(super.rootContext)
   
    private def addMagicImports(initCtx: Context): Context = ...
  }
}
```
This way I wouldn't need to modify this line